workloads/librispeech/librispeech_jax/models.py

```diff
@@ -0,0 +1,257 @@
+  train: bool = False
+
+  def setup(self):
+    if self.use_batch_norm:
+      self.batch_norm = SequenceWise(nn.BatchNorm(use_running_average=True))
+    else:
+      self.batch_norm = None
+    self.rnn = SimpleBiLSTM(self.hidden_size)
+
+  def __call__(self, inputs, lengths, training=False):
+    if self.batch_norm is not None:
+      inputs = self.batch_norm(inputs)
+    inputs = inputs.transpose(1, 0, 2)
+    inputs = self.rnn(inputs, lengths)
+    inputs = inputs.transpose(1, 0, 2)
+    # (TxNxH*2) -> (TxNxH) by sum
+    inputs = inputs.reshape(inputs.shape[0], inputs.shape[1], -1).sum(2)
+    return inputs
+
+
+def hard_tanh(x, min_value=-1., max_value=1.):
+
+  return jnp.where(x > max_value, max_value, jnp.where(x < min_value, min_value, x))
+
+
+def get_seq_lens(input_length, conv_seq_module):
+    """Get a 1D tensor or variable containing the size sequences that will be output by the network.
+
+    Args:
+      input_length: 1D Tensor
+
+    Returns:
+      1D Tensor scaled by model
+    """
+    seq_len = input_length
+    for m in conv_seq_module:
+      if isinstance(m, nn.Conv):
+        seq_len = (seq_len + 2 * m.padding[1][1] - m.kernel_dilation * (m.kernel_size[1] - 1) - 1) // m.strides[1] + 1
+    return seq_len
+
+
+class CNNLSTM(nn.Module):
+
+  num_classes = 29
+  hidden_size = 768
+  hidden_layers = 5
+  context = 20
+
+  def setup(self):
+    sequential = [
+      nn.Conv(
+        features=32,
+        kernel_size=(41, 11),
+        strides=(2, 2),
+        padding=((20, 20), (5, 5)),
+      ),
+      nn.BatchNorm(use_running_average=True),
+      functools.partial(hard_tanh, min_value=0, max_value=20),
+      nn.Conv(
+        features=32,
+        kernel_size=(21, 11),
+        strides=(2, 1),
+        padding=((10, 10), (5, 5)),
+      ),
+      nn.BatchNorm(use_running_average=True),
+      functools.partial(hard_tanh, min_value=0, max_value=20),
+    ]
+    self.conv = MaskConv(sequential)
+
+    rnn_input_size = 161
+    rnn_input_size = int(math.floor(rnn_input_size + 2 * 20 - 41) / 2 + 1)
+    rnn_input_size = int(math.floor(rnn_input_size + 2 * 10 - 21) / 2 + 1)
+    rnn_input_size *= 32
+
+    rnns = []
+    rnn = BatchRNN(
+        input_size=rnn_input_size,
+        hidden_size=self.hidden_size,
+        use_batch_norm=False,
+    )
+    rnns.append(rnn)
+    for _ in range(self.hidden_layers - 1):
+      rnn = BatchRNN(
+          input_size=self.hidden_size,
+          hidden_size=self.hidden_size
+          )
+      rnns.append(rnn)
+
+    self.rnns = rnns
+
+    fully_connected = Sequential([
+      nn.BatchNorm(use_running_average=True),
+      nn.Dense(self.num_classes, use_bias=False)
+    ])
+
+    self.fc = SequenceWise(fully_connected)
+
+  def __call__(self, inputs, lengths, training=False):
+    output_lengths = get_seq_lens(lengths, self.conv.seq_module)
+
+    x, _ = self.conv(inputs, lengths)
+
+    sizes = x.shape
+    x = x.reshape(sizes[0], sizes[1] * sizes[2],
+               sizes[3]) 
```

Comment:
can all of these `sizes` be on the same line?

Reply:
done